### PR TITLE
[integration] move ConfReference DIRAC section to dirac.cfg

### PR DESCRIPTION
**Pull request by TaykYoku**
Comes from https://github.com/DIRACGrid/DIRAC/pull/5397#discussion_r742955718. 
This PR wants to move the `ConfReference` DIRAC section part to `dirac.cfg` and add there OAuth 2 settings.

BEGINRELEASENOTES

*docs
CHANGE: move the ConfReference DIRAC section part to dirac.cfg and add there OAuth 2 settings.

ENDRELEASENOTES

### CODE REVIEW

**Review by andresailer — COMMENTED** [review]
Many of the comment lines are too long and require scrolling, which makes this very hard to read.
https://dirac--5573.org.readthedocs.build/en/5573/AdministratorGuide/Configuration/ExampleConfig.html

File: dirac.cfg [hunk 1]
```diff
@@ -1,3 +1,116 @@
...
+    #CertFile = /opt/dirac/etc/grid-security/hostcert.pem
+
+    # Directory where host key is located in the server. If not specified, DIRAC will try to find it.
+    #KeyFile = /opt/dirac/etc/grid-security/hostkey.pem
+
+    # Flag to use access tokens and not user proxies. This is typically false.
+    #UseTokens = true
+
+    # Section that describe OAuth 2.0 authorization settings and metadata. Look https://datatracker.ietf.org/doc/html/rfc8414.
+    # In most of these metadata is filled in automatically.
```
[the remark this answers is not in the document]
Comment (TaykYoku) [reply dirac.cfg:69]: I mean OAuth 2.0 Authorization Server Metadata. And that you need to determine only the issuer metadata name, the value of other names will automatically determine the code, referring to the appropriate RFC (https://datatracker.ietf.org/doc/html/rfc8414#section-2)

File: dirac.cfg [hunk 2]
```diff
@@ -1,3 +1,116 @@
...
+    Authorization
+    {
+      # The authorization server's issuer identifier, which is a URL that uses the "https" scheme and has no query or fragment components. Look https://datatracker.ietf.org/doc/html/rfc8414#section-2.
+      # This option is described in the global configuration and when the DIRAC client is installed and is required to interact with the DIRAC Authorization Server.
```
[the remark this answers is not in the document]
Comment (TaykYoku) [reply dirac.cfg:73]: DIRAC has several levels of configuration, as described here https://dirac.readthedocs.io/en/latest/AdministratorGuide/Configuration/ConfigurationStructure/index.html#configuration-sources.
I mean, these settings should be synchronized with `DIRAC CS`, not just specified in the local `dirac.cfg` files. Also when installing the client, this section must be present in `dirac.cfg`.

File: dirac.cfg [hunk 3]
```diff
@@ -1,3 +1,116 @@
...
+    # Directory where host certificate is located in the server. If not specified, DIRAC will try to find it.
+    #CertFile = /opt/dirac/etc/grid-security/hostcert.pem
+
+    # Directory where host key is located in the server. If not specified, DIRAC will try to find it.
+    #KeyFile = /opt/dirac/etc/grid-security/hostkey.pem
+
+    # Flag to use access tokens and not user proxies. This is typically false.
+    #UseTokens = true
+
+    # Section that describe OAuth 2.0 authorization settings and metadata. Look https://datatracker.ietf.org/doc/html/rfc8414.
```
[the remark this answers is not in the document]
Comment (TaykYoku) [reply dirac.cfg:68]: fixed

File: dirac.cfg [hunk 4]
```diff
@@ -1,3 +1,116 @@
...
+    # Set slaves grace time in a seconds. By default 600.
+    #SlavesGraceTime = 600
+
+    # CS configuration version used by DIRAC services as indicator when they need to reload the
+    # configuration. Expressed using date format. By default 0.
+    #Version = 2011-02-22 15:17:41.811223
+  }
+
+  # This subsection defines several options related to the DIRAC security framework
+  Security
```
[the remark this answers is not in the document]
Comment (TaykYoku) [reply dirac.cfg:52]: added

File: dirac.cfg (hunk 2 above)
[the remark this answers is not in the document]
Comment (TaykYoku) [reply dirac.cfg]: Given the warning I added about the Security section, I paraphrased this line.

File: dirac.cfg (hunk 1 above)
[the remark this answers is not in the document]
Comment (TaykYoku) [reply dirac.cfg]: I paraphrased this line.

**Review by fstagni — APPROVED** [review]

File: dirac.cfg [hunk 5]
```diff
@@ -1,3 +1,115 @@
...
+    #KeyFile = /opt/dirac/etc/grid-security/hostkey.pem
+
+    # Flag to use access tokens and not user proxies. This is typically false.
+    #UseTokens = true
+
+    # Section that describes OAuth 2.0 authorization settings and metadata, it is required to interact with the DIRAC Authorization Server. Please see https://datatracker.ietf.org/doc/html/rfc8414 for more details.
+    Authorization
+    {
+      # The authorization server's issuer identifier, which is a URL that uses the "https" scheme and has no query or fragment components. Please see https://datatracker.ietf.org/doc/html/rfc8414#section-2.
+      # This option must be defined in the client installations and on the DIRAC Authorization Server host, based on it other Authorization Server metadata names can be defined automatically.
```
Comment (andresailer) [inline dirac.cfg:73]:
```suggestion
      # This option must be defined in the client installations and on the DIRAC Authorization Server host.
```
Do I understand correctly, that the Issue URL is used inside the DIRAC (or OAuth) code to extract other information? Then I don't think you need to mention this here.

File: dirac.cfg (hunk 1 above)
[the remark this answers is not in the document]
Comment (andresailer) [reply dirac.cfg]: I saw in the rfc link that it is called metadata there, which I find weird, but OK to follow their diction in that case.

File: dirac.cfg (hunk 5 above)
Comment (TaykYoku) [reply dirac.cfg:73]: First running the `DIRAC Authorization Server` (AS), it reads the `DIRAC/Security/Authorization` section and subsections to generate its metadata according to the **RFC**. Then when the `DIRAC client` is installed, this metadata is read from the **_https://<DIRAC AS issuer>/.well-known/oauth-authorization-server_** request (https://datatracker.ietf.org/doc/html/rfc8414#section-3.1).

Comment (TaykYoku) [reply dirac.cfg:73]: > based on it other Authorization Server metadata names can be defined automatically

means that when DIRAC AS generates its metadata, DIRAC AS issuer enough to do so

Comment (andresailer) [reply dirac.cfg:73]: So this phrase can be dropped?

Comment (TaykYoku) [reply dirac.cfg:73]: This is mentioned in the comments, code and documentation, so I think it can be removed here.

**Comment by TaykYoku** [comment]
> Many of the comment lines are too long and require scrolling, which makes this very hard to read.

I think this will be difficult to avoid, as we decided to keep a description of all sections and options here. What solution do you see?

If dirac.cfg becomes a guide or something similar, I think the administrator who is just familiar with the DIRAC configuration will be happy to see a more detailed description.

**Comment by TaykYoku** [comment]
> Many of the comment lines are too long

made them shorter

**Review by fstagni — APPROVED** [review]

**Review by andresailer — APPROVED** [review]
Regarding the BEGIN/ENDRELEASENOTES in the PR opening comment.

You shouldn't put everything inside these "Tags" only the *System\nFIX/NEW/CHANGE parts.

**Comment by TaykYoku** [comment]
Thanks for this note, I moved all descriptions outside BEGIN/ENDRELEASENOTES tags.

**Comment by TaykYoku** [comment]
IMHO its ready to merge.

**Pull request merged**